modules/collabora_online_group/tests/src/Kernel/AccessTest.php

```diff
@@ -0,0 +1,127 @@
+     *
+     * @dataProvider dataProvider
+     */
+    public function testCollaboraAccess(bool $expected_result, array $permissions, array $group_permissions, string $operation, string $scope = ''): void {
+        // Create group type, media type and enable plugin.
+        $group_type = $this->createGroupType();
+        $this->createGroupRole([
+            'group_type' => $group_type->id(),
+            'scope' => PermissionScopeInterface::INSIDER_ID,
+            'global_role' => RoleInterface::AUTHENTICATED_ID,
+            'permissions' => $group_permissions
+        ]);
+        $this->createMediaType('file', ['id' => 'document']);
+        $this->createPluginRelation($group_type, 'group_media:document', [
+            'group_cardinality' => 0,
+            'entity_cardinality' => 1,
+            'use_creation_wizard' => FALSE,
+        ]);
+
+        // Add group, media and relation between both.
+        $group = $this->createGroup(['type' => $group_type->id()]);
+        $media = $this->createMediaEntity('document');
+        $group->addRelationship($media, 'group_media:document');
+
+        // Create the user with the given permissions and as member of the group.
+        $user = $this->createUser($permissions);
+        $group->addMember($user);
+
+        if ($scope === 'own') {
+            $media->setOwnerId($user->id())->save();
+        }
+
+        // Check access.
+        $this->assertEquals($expected_result, $media->access($operation, $user));
+   }
+
+    /**
+     * Data provider.
+     *
+     * @return \Generator
+     *   The test data.
+     */
+    protected function dataProvider(): \Generator {
+        // Preview: user with no permissions at all.
+        yield [FALSE, [], [], 'preview in collabora'];
+        // Preview: user with global permissions doesn't have access.
+        yield [FALSE, ['preview document in collabora'], [], 'preview in collabora'];
+        // Preview: user with group permissions have access.
+        yield [TRUE, [], ['preview group_media:document in collabora'], 'preview in collabora'];
+        // Edit any: user with no permissions at all.
+        yield [FALSE, [], [], 'edit in collabora'];
+        // Edit any: user with global permissions doesn't have access.
+        yield [FALSE, ['edit any document in collabora'], [], 'edit in collabora'];
+        // Edit any: User with group permissions have access.
+        yield [TRUE, [], ['edit any group_media:document in collabora'], 'edit in collabora'];
+        // Edit own: user with no permissions at all.
+        yield [FALSE, [], [], 'edit in collabora', 'own'];
+        // Edit own: user with global permissions doesn't have access.
+        yield [FALSE, ['edit own document in collabora'], [], 'edit in collabora', 'own' ];
+        // Edit own: user with group permissions have access.
+        yield [TRUE, [], ['edit own group_media:document in collabora'], 'edit in collabora', 'own'];
+   }
```

Comment:
I know this pattern is found in core, and probably in other places too. So I cannot say it is "wrong".
The problem is that we have a separate Drupal install per dataset, which is slow.
I don't think we gain that much from the isolation.

If you look at https://github.com/donquixote/collabora-drupal/pull/12/files, there is a different way to test access with only one test run. I won't say that is the "correct" way of doing it, let me know what you think.

Reply:
Aside of that, I was thinking we should add string keys for the dataset.
But now I am changing my mind: The test output might be more readable without the string keys.

Reply:
> If you look at https://github.com/donquixote/collabora-drupal/pull/12/files, there is a different way to test access with only one test run.

One way to do this here would be:
- Create just one group type and group (I think is enough).
- Create multiple media types and relationship types, and the respective media entities.
- Create multiple group roles, to cover different combinations of group permissions, and then build an access matrix with user + media + operation.
- Create users with the different roles.
- Build and assert an access matrix of user + media entity + operations they can access.

But, we can also keep this test as is for now, and consider to change it later, so we can move on.

Reply:
Yes, it's true that the test takes a lot of time, converted the PHPUnit data provider to an array based test.

Reply:
Execution went from 30s to 3s locally.

Reply:
btw, about the string keys, I initially had something different in mind :)

```php
            yield 'Preview: user with no permissions at all' => [...],
            yield '...' => [...],
            yield '...' => [...],
```

But the string keys in the data itself is also not bad.

Reply:
Especially now it would be useful to have these dataset names, and use them in the assertion, so that we know which dataset is failing.